thinc/schedules.py

```diff
@@ -1,38 +1,81 @@
 """Generators that provide different rates, schedules, decays or series."""
-from typing_extensions import Protocol, runtime_checkable
+from typing import Any, Callable, Dict, Generic, TypeVar
 import numpy
 
 from .config import registry
-from .types import ScheduleCallable
+
+OutT = TypeVar("OutT")
+
+
+class Schedule(Generic[OutT]):
+    """Class for implementing Thinc schedules."""
+
+    name: str
+    _schedule: Callable
+    _attrs: Dict[str, Any]
+
+    __slots__ = ["name", "_schedule", "_attrs"]
+
+    def __init__(
+        self, name: str, schedule: Callable, *, attrs: Dict[str, Any] = {}
```

Comment:
Is it a worry that `attrs` has a mutable default value here? We can also set it to `None` and rewrite to `{}` within the function.

Reply:
I did the same as `Model` does, make a copy of the default value in the constructor. I think that should cover it?